### ISSUE TITLE
OFTCPSocket asyncConnect crash

### ISSUE THREAD

**Issue by GunGraveKoga** [Midar/objfw#26]
Very often crashes when using asynchronous connection to remote host.
If exception has occurred, I try to write error information about it in log file in an another thread. As result I got crash like this:
6F1646D6 02475820 00000003 00000003  libobjfw.dll!objc_exit
004019CA 02478CB8 02478CB8 02478CB8  test.exe!__destroy_helper_block_
6F08BC24 02478CB8 6F19ADD0 02478CB8  libobjfw.dll!_Block_release
6F08C376 02478CB8 6F19ADD0 00186B90  libobjfw.dll!_i_OFBlock__release
6F0FAACF 02472548 6F18F110 02472548  libobjfw.dll!_i_OFTimer__dealloc
6F0CAD8E 02472548 6F1B20D0 00000000  libobjfw.dll!_i_OFObject__release
6F15DCDF 00000000 6F191B70 00000000  libobjfw.dll!objc_autoreleasePoolPop
6F0D67D0 0246F540 6F191B68 00000000  libobjfw.dll!_i_OFRunLoop__runUntilDate_
6F0D5CF6 0246F540 6F1818B8 0246F540  libobjfw.dll!_i_OFRunLoop__run
6F0829F9 0216D208 6F181900 02478D08  libobjfw.dll!_i_OFApplication__OF_run
6F0818A1 0061FEAC 0061FEA8 00403C70  libobjfw.dll!of_application_main
0040167E 00000001 0216EF88 02161E30  test.exe!main
004013DE 00287000 759C38D0 B466C8F1  test.exe!__tmainCRTStartup
759C38F4 00287000 B6463519 00000000  KERNEL32.DLL!BaseThreadInitThunk
777C5DE3 FFFFFFFF 777EB7E6 00000000  ntdll.dll!__RtlUserThreadStart
777C5DAE 004014E0 00287000 00000000  ntdll.dll!_RtlUserThreadStart

Simple code:
```
#import<ObjFW/ObjFW.h>

@interface Delegate: OFObject <OFApplicationDelegate>

- (void)applicationDidFinishLaunching;
- (void)connect:(OFTCPSocket *)s except:(OFException *)e;

@end

OF_APPLICATION_DELEGATE(Delegate)


@implementation Delegate

- (void)applicationDidFinishLaunching
{
        OFTCPSocket* sock = [OFTCPSocket socket];
	[sock asyncConnectToHost:@"localhost" port:25 target:self selector:@selector(connect:except:)];
}

- (void)connect:(OFTCPSocket *)s except:(OFException *)e
{
	if (e != nil) {
		OFTimer* tm = [OFTimer timerWithTimeInterval:0.0 repeats:false block:^(OFTimer *timer){
				of_log(@"%@", e);
			
		}];

		[[OFRunLoop mainRunLoop] addTimer:tm];
		return;
	}
}

@end
```
[crash.txt](https://github.com/Midar/objfw/files/205373/crash.txt)

**Comment by Midar** [comment]
I don't get any output when running this and no exception either, which is to be expected, as there's only code handling the exception. The stack trace you provide unfortunately does not contain any source lines, so it's hard to follow what happened.

**Comment by GunGraveKoga** [comment]
OK, with latest commits of objfw I unable to reproduce  this bug
I get output with exception description on every execution

**Comment by Midar** [comment]
So, it is working as intended?

**Comment by GunGraveKoga** [comment]
Yes

**Comment by Midar** [comment]
Ok, closing this then. Please reopen again if it should ever happen again. For now, I'm closing it, as it does not seem to be reproducible for anyone anymore.
